runtime/src/main/java/org/corfudb/runtime/clients/BaseClient.java

```diff
@@ -182,4 +183,13 @@ private static Object handleNotReady(CorfuMsg msg, ChannelHandlerContext ctx, IC
         throw new ServerNotReadyException();
     }
 
+    /** Generic handler for a server exception. */
+    @ClientHandler(type = CorfuMsgType.ERROR_SERVER_EXCEPTION)
+    private static Object handleServerException(CorfuPayloadMsg<ExceptionMsg> msg,
+                                                ChannelHandlerContext ctx, IClientRouter r)
+        throws Throwable {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)